data_steward/tools/clean_test_project.py

```diff
@@ -0,0 +1,107 @@
+# Python imports
+import argparse
+import logging
+import sys
+
+# Third party imports
+from googleapiclient.errors import HttpError
+
+# Project imports
+from utils import bq
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+
+
+def get_datasets_with_substrings(datasets_list, name_substrings):
+    """
+    Filters list of datasets with specified substrings (e.g. github usernames) in them
+
+    :param datasets_list: list of dataset_ids
+    :param name_substrings: identifies substrings that help identify datasets to delete
+    :return: list of dataset_ids with any substring in their dataset_id
+    """
+    datasets_with_substrings = []
+    for dataset in datasets_list:
+        if any(name_substring in dataset for name_substring in name_substrings):
+            datasets_with_substrings.append(dataset)
+    return datasets_with_substrings
+
+
+def delete_datasets(project_id, datasets_to_delete_list):
+    """
+    Deletes datasets using their dataset_ids
+
+    :param project_id: identifies the project
+    :param datasets_to_delete_list: list of dataset_ids to delete
+    :return:
+    """
+    failed_to_delete = []
+    for dataset in datasets_to_delete_list:
+        try:
+            bq.delete_dataset(project_id, dataset)
+            logging.info(f'Deleted dataset {dataset}')
+        except HttpError:
+            logging.exception(f'Could not delete dataset {dataset}')
+            failed_to_delete.append(dataset)
+    logging.info(
+        f'The following datasets could not be deleted: {failed_to_delete}')
+
+
+def run_deletion(project_id, name_substrings):
+    """
+    1. Retrieves all datasets in project,
+    2. Filters the list based on criteria (eg. github username to identify circle datasets)
```

Comment:
Please add a one line summary followed by a blank line.  Then the rest of your doc string is good.